core/lib/dal/src/models/storage_transaction.rs

```diff
@@ -336,8 +337,14 @@ impl From<StorageApiTransaction> for api::Transaction {
 impl<'r> FromRow<'r, PgRow> for StorageApiTransaction {
     fn from_row(db_row: &'r PgRow) -> Result<Self, Error> {
         let row_signature: Option<Vec<u8>> = db_row.get("signature");
-        let signature = row_signature
-            .and_then(|signature| PackedEthSignature::deserialize_packed(&signature).ok());
+        let transaction_type = db_row
+            .try_get::<Option<i32>, &str>("tx_format")
+            .unwrap_or_default();
+
+        let (mut v, mut s, mut r) = (None, None, None);
+        if let (Some(signature), Some(tx_type)) = (&row_signature, transaction_type) {
+            (v, s, r) = signature_to_vrs(signature, tx_type as u32);
+        }
```

Comment:
I think it will be a bit cleaner. 

```suggestion
        let (v,r,s) = if let (Some(signature), Some(tx_type)) = (&row_signature, transaction_type) {
             signature_to_vrs(signature, tx_type as u32)
        } else {
             (None, None, None)
        }
```

Reply:
And even with this suggestion, you messed up params. 
you have (v,s,r), but function returns (v,r,s)